### ISSUE TITLE
how to train myself codebook?

### ISSUE THREAD

**Issue by wxde** [rmurai0610/MASt3R-SLAM#80]
hi, thanks for your work, if i want to use myself image, how to get the codebook? parameter P and  and M in the class Whiten, how to set the loss to train the parameter?

**Comment by edexheim** [comment]
Hi,

We did not train the codebook, but used the one from [MASt3R-SfM](https://github.com/naver/mast3r/tree/mast3r_sfm) with some modifications to run the database construction and retrieval incrementally and online.  It seems to be set up as a Pytorch module, but I am not sure if the training code/data is released.

Best,
Eric

**Comment by wxde** [comment]
thanks
